### PR TITLE
Tidy up a bit and fix a few narrowing warnings

### PR DESCRIPTION
**Pull request by bkentel**
**options.h**
- Flip c-style enum to c++.
- Remove redundant forward declaration.

**overmap.***
- Cleanup note related functions
- Remove unused field from `om_note`.
- Remove constructor from `om_note`.
- Make `nullstr` a function local static.
- Fix a few narrowing warnings by using an integer multiply then divide.
- Use in-class initializers for `groundcover_extra`.
- Use in-class initializers for `regional_settings`.
- Remove some redundant const.

### CODE REVIEW

File: src/overmap.cpp [hunk 1]
```diff
@@ -713,52 +712,52 @@ bool overmap::has_note(int const x, int const y, int const z) const
         return false;
     }
 
-    for (auto &i : layer[z + OVERMAP_DEPTH].notes) {
```
Comment (kevingranade) [inline src/overmap.cpp]: The logic is equivalent, and merely switches to an idiom with no benefits in this case.

File: src/overmap.cpp [hunk 2]
```diff
@@ -713,52 +712,52 @@ bool overmap::has_note(int const x, int const y, int const z) const
...
-                message));
+    auto &notes = layer[z + OVERMAP_DEPTH].notes;
+    auto const it = std::find_if(begin(notes), end(notes), [&](om_note const& n) {
+        return n.x == x && n.y == y;
+    });
+
+    if (message.empty() && it != notes.end()) {
+        notes.erase(it);
+    } else if(!message.empty()) {
+        notes.emplace_back(om_note {std::move(message), x, y});
```
Comment (BevapDin) [inline src/overmap.cpp]: Doesn't that change the logic (haven't tested) when this is called with a non-empty message on a place that already has a note? In that case `it != end`, but message is non-empty, so a *new* note is *added*, the previous code would instead change the existing entry.

Comment (bkentel) [inline src/overmap.cpp]: @BevapDin Yep. I'll take care of that.

File: src/overmap.cpp (hunk 1 above)
Comment (bkentel) [inline src/overmap.cpp]: Fair enough. I tend to think that being explicit with a named algorithm is better though.
A wonderful [talk](http://channel9.msdn.com/Events/GoingNative/2013/Cpp-Seasoning) given about a year ago on the matter. A good watch regardless of this particular case.

**Pull request merged**